### DISCUSSION TITLE
Device explorer is unable to find a connected device in VS 2022 version 17.5.4

### DISCUSSION THREAD

**Discussion by jasonkaisersmith** [nanoframework/Home#1279] [Q&A] [answered]
### Tool

Visual Studio extension

### Description

Device explorer is unable to find a connected device.

Output in VS
07:07:10.504 [PortSerial device watcher stopped @ Thread 83]
07:07:10.519 [PortSerial device watcher started @ Thread 82 [ProcessID: 12896]]
07:07:10.520 [NanoDevices: new device arrival COM6]
07:07:10.520 [NanoDevices: dropping device in exclusion list COM4]
07:07:10.520 [NanoDevices: candidate nano device COM6]
07:07:10.520 [NanoDevices: dropping device in exclusion list COM3]
07:07:10.531 [NanoDevices: checking device  COM6 @ 921600]
07:07:11.555 [NanoDevices: checking device  COM6 @ 460800]
07:07:12.579 [NanoDevices: checking device  COM6 @ 115200]
07:07:15.545 [NanoDevices: checking device  COM6 *** 2nd attempt ***]
07:07:15.561 [NanoDevices: checking device  COM6 @ 921600]
07:07:17.596 [NanoDevices: checking device  COM6 @ 460800]
07:07:19.620 [NanoDevices: checking device  COM6 @ 115200]
07:07:22.257 [NanoDevices: quitting device COM6]
07:07:22.257 [NanoDevices: Serial device enumeration completed. Found 0 devices]

Device is visible in device manager (no errors) on Win10.

nanoff can view the device details with no issue.

nanoff --serialport COM6 --devicedetails
.NET nanoFramework Firmware Flasher v2.4.33+4ca230f662
Copyright (C) 2019 .NET Foundation and nanoFramework project contributors

Using COM6 @ 1500000 baud to connect to ESP32.
Reading details from chip...
OK

Connected to:
ESP32 (ESP32-D0WD-V3 (revision 3))
Features WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Flash size 4MB unknown from TENX (manufacturer 0x94 device 0x16406)
PSRAM: not available
Crystal 40MHz
MAC B0:B2:1C:97:8D:20

### Aditional context

VS 2022 version 17.5.4
Extension version 2022.3.0.41

It worked previously in VS 2022 VS version 17.5.3

**Comment by josesimoes** [comment, marked as answer]
Can you please give it a try with `nanoff --update --platform esp32 --serialport COM6` ? 
Then unplug and plug it again.

**Comment by renerlemes** [comment]
I did this procedure and it still doesn't show in Device Explorer. My Visual Studio is in version 17.6.2 and when it was in version 17.5.3 it worked too.

**Reply by josesimoes** [reply]
There where no changes in VS extension that could cause this.
Most likely the firmware is not the appropriate one or other application is blocking the access to the COM port.

**Reply by renerlemes** [reply]
@josesimoes I used nanoff with the command you provided and also did a clean install of VS and the Nanoframework extension. It didn't work either! Strange that it detects the board, it is possible to use the nanoff commands, only in VS that does not work.

**Reply by josesimoes** [reply]
VS and nanoff use different ways of communicating with ESP32 devices, therefore the difference in behaviour.

**Reply by josesimoes** [reply]
@renerlemes are you on our  Discord server? If yes, that's a much better channel to ask & provide support.

**Comment by jasonkaisersmith** [comment]
One thing that has worked for me, when I have strange problems, is to erase flash on the the device Using https://github.com/espressif/esptool (of course this will depend on what board you have, so take care!)

`esptool.py --port com6 erase_flash`

And then install again via nanoff.

Make sure you restart VS before you do this, just to be sure.

**Comment by NimeCloud** [comment]
In my case I need to press the download mode button during the scan.
